pytorch_lightning/strategies/horovod.py

```diff
@@ -184,8 +196,13 @@ def post_backward(self, closure_loss: torch.Tensor) -> None:
 
     def _wrap_optimizers(self, optimizers: List[Optimizer]) -> List["hvd.DistributedOptimizer"]:
         """Wraps optimizers to perform gradient aggregation via allreduce."""
+        accumulate_grad_batches = self.lightning_module.trainer.accumulate_grad_batches
```

Comment:
pass accumulate_grad_batches as an argument to this function from the `setup` function above. it's easier to test this function in isolation then

Reply:
Thanks, @ananthsub for taking a look. I have made the changes, but curious to know what you feel about the tests added in this PR?